src/backend_model_instance.cc

```diff
@@ -247,6 +253,21 @@ TritonModelInstance::CreateInstances(
         const auto& kind = std::get<1>(is);
         const auto& id = std::get<2>(is);
 
+        const Signature signature(group, id);
+        // Check if an existing instance can be re-used.
+        if (!ShareBackendThread(model->DeviceBlocking(), kind)) {
+          auto existing_instance = model->FindInstance(signature);
```

Comment:
Add the link to the jira ticket to improve the search process as a TO-DO.

Reply:
One solution could be to use hash key.

Reply:
Updated: [Add links to JIRA](https://github.com/triton-inference-server/core/pull/189/commits/9f65d1267f872ff863c7b0d66910a159e6a3f561)